src/core/eventhandler.rs

```diff
@@ -111,9 +114,53 @@ impl EventHandler for Handler {
       .await
       .unwrap()
       .is_own(&ctx.cache);
+    let user_id = reaction.user_id.unwrap();
+    let guild = parse::get_guild(reaction.channel_id, &ctx, None)
+      .await
+      .unwrap();
 
-    if reaction.user_id.unwrap() != botid && isown {
+    if user_id != botid && isown {
       let emoji = reaction.emoji.as_data();
+
+      if let Ok(member) = guild.member(&ctx.http, user_id).await {
```

Comment:
I would like that you re-use the system below, you already have emoji validation, and doing the check of the emoji before looking if it's a message that you are tracking is really a lot less expensive.

\+ you can remove the:
```rs
  pub accept: ReactionType,
  pub reject: ReactionType,
```

Reply:
![image](https://github.com/blackfoot-makers/bf-discord-rbot/assets/33382871/91a017d8-8a42-4817-9dc7-9966a2ef6381)

![image](https://github.com/blackfoot-makers/bf-discord-rbot/assets/33382871/517d814f-185d-4d03-bf22-050b218b8452)


Want me to improve your "system below" ? Because the "system below" is not efficient too it should be called before or after the `match` a perform an early return because you don't want to continue if it match